### ISSUE TITLE
Need support for exact string match for path in routes.json

### ISSUE THREAD

**Issue by saraxu** [patrick-steele-idem/meta-router#9]
I have a rule defined in route.json as below, but it's triggered by a page url: `http://localhost:8080/growth/foo`, this url should be redirect to a 404 page.

```json
{
    "path": "GET /growth",
    "handler": "require:./src/pages/recommendations",
    "pagename": "growth_recommendations"
 }
```

**Comment by patrick-steele-idem** [comment]
I was able to reproduce this problem with a test case. The problem is in the following code:
https://github.com/patrick-steele-idem/meta-router/blob/be9e04a999b3526ac3f99f2d34f31e8847c8f6a4/lib/Route.js#L71-L75

We are currently generating the `RegExp` using the following options:

```javascript
{
    sensitive: routeConfig.caseSensitive === true,
    strict: routeConfig.strict === true,
    end: routeConfig.end === true
}
```

This is default `end` to `false`, when it should be defaulting to `true`.

The following code would match Express by default:

```javascript
{
    sensitive: routeConfig.caseSensitive === true,
    strict: routeConfig.strict === true,
    end: routeConfig.end !== false
}
```

See: https://github.com/component/path-to-regexp/blob/4c5412af6fae141f48c32e535bc931573ade99c4/index.js#L71

I'll update soon with suggestions for a fix. I'm leaning towards releasing a new major version with slightly different behavior for configuring how routes are matched.

**Comment by patrick-steele-idem** [comment]
Fixed by 27df85e4fb1012e88e43363fb65c0ebb8e889d0a

New version published: `meta-router@3.0.0`

NOTE: This was a major version increase. You must install the new version by doing the following:

```
npm install meta-router@^3 --save
```

Please let me know if you see any issues. Thanks for reporting the problem.

**Comment by patrick-steele-idem** [comment]
Hi @saraxu, this was a major version update, but if you just upgrade to the latest meta-router your problem should be resolved and it is unlikely that your code will need to be changed. This was a major version upgrade since the default matching options were changed to match what Express uses. Previously, meta-router would match only the beginning by default, but now it matches to the end by default.
